### DISCUSSION TITLE
关于预测天数的疑问

### DISCUSSION THREAD

**Discussion by darryhh** [Monaco12138/Convlstm#5] [Q&A] [answered]
在
/model_factory.py第49行，loss = self.MSE_criterion(next_frames, frames_tensor[:, 1:])，您注释也写了：是两个形状为(4 , 19 , 140//4 , 140//4 , 4*4*1)的数组计算loss。既然y（标签）和网络输出的时间步长都是19，那是不是可以理解为您预测了19天。

我个人一点浅见：如果预测十天，那应该只把训练数据的前10个时间步(4 , 10, 140//4 , 140//4 , 4*4*1)输入进网络，这样网络输出的shape也是(4 , 10, 140//4 , 140//4 , 4*4*1)，最后将训练数据的后10个时间步和网络输出计算loss。

期待得到您的指点

**Comment by Monaco12138** [comment, marked as answer]
您可以看一下./models/convlstm.py文件中的第48到60行，实际上模型训练过程中，一开始就让它预测学习太长的天数会不稳定，所以融合了一些真实的信息，在最开始的几轮训练中，mask_true全部设置为1，即让1-19帧去预测2-20。随着训练的轮数增加，会将mask_true设置为0，此时模型的输入20帧并没有给我们全部用上，我们只用了1-10帧，后面的信息依次都是那前面的预测值作为输入的。
之所以让输入是20帧，有两点原因，第一是为了在训练一开始时融合一些真实的信息让训练效果更好。第二是在测试时，可以直接用后10帧作为label算误差。
